include/ginkgo/core/solver/chebyshev.hpp

```diff
@@ -0,0 +1,236 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#ifndef GKO_PUBLIC_CORE_SOLVER_CHEBYSHEV_HPP_
+#define GKO_PUBLIC_CORE_SOLVER_CHEBYSHEV_HPP_
+
+
+#include <vector>
+
+#include <ginkgo/core/base/exception_helpers.hpp>
+#include <ginkgo/core/base/lin_op.hpp>
+#include <ginkgo/core/base/types.hpp>
+#include <ginkgo/core/config/config.hpp>
+#include <ginkgo/core/config/registry.hpp>
+#include <ginkgo/core/matrix/dense.hpp>
+#include <ginkgo/core/matrix/identity.hpp>
+#include <ginkgo/core/solver/solver_base.hpp>
+#include <ginkgo/core/stop/combined.hpp>
+#include <ginkgo/core/stop/criterion.hpp>
+#include <ginkgo/core/stop/iteration.hpp>
+
+
+namespace gko {
+namespace solver {
+
+
+/**
+ * Chebyshev iteration is an iterative method that can solving nonsymeetric
+ * problems. Chebyshev Iterations avoids the inner products for computation
+ * which may be the bottleneck for distributed system. Chebyshev Iteration is
+ * developed based on the Chebyshev polynomials of the first kind. Moreover,
+ * this method requires knowledge about the spectrum of the preconditioned
+ * matrix. This implementation follows the algorithm in "Templates for the
+ * Solution of Linear Systems: Building Blocks for Iterative Methods, 2nd
+ * Edition".
```

Comment:
I would rearrange this slightly (the fact that we need to know the spectrum is important, so we should mention it early), and try to be consistent with capitalization of iteration.
```suggestion
 * Chebyshev iteration is an iterative method for solving nonsymmeetric
 * problems based on some knowledge of the spectrum of the (preconditioned) system matrix. It avoids the computation of inner products,
 * which may be a performance bottleneck for distributed system. Chebyshev iteration is
 * developed based on Chebyshev polynomials of the first kind. 
 * This implementation follows the algorithm in "Templates for the
 * Solution of Linear Systems: Building Blocks for Iterative Methods, 2nd
 * Edition".
```